workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/JiraWebService.java

```diff
@@ -0,0 +1,352 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.customer.service;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.Base64;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * @author Jenny Chen
+ */
+@Component
+public class JiraWebService {
+
+	public String getJiraIssue(String issueKey) throws Exception {
+		StringBundler sb = new StringBundler(3);
+
+		sb.append(_URL_REST_API_2);
+		sb.append("/issue/");
+		sb.append(issueKey);
+
+		try {
+			JSONObject jsonObject = new JSONObject(
+				WebClient.create(
+					_jiraURL
+				).get(
+				).uri(
+					sb.toString()
+				).accept(
+					MediaType.APPLICATION_JSON
+				).header(
+					HttpHeaders.AUTHORIZATION, _getCredentials()
+				).retrieve(
+				).bodyToMono(
+					String.class
+				).block());
+
+			jsonObject = _processResponse(jsonObject);
+
+			return jsonObject.toString();
+		}
+		catch (Exception exception) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					"Unable to fetch Jira issue with key " + issueKey,
+					exception);
+			}
+		}
+
+		return null;
+	}
+
+	public String getJiraSearch(String keywords, String role) throws Exception {
```

Comment:
thinking ahead to having to build changelog with this web service as well in the very near future, would it be worth it to abstract out the generic search logic a bit more and separate it from the security vulnerability specific logic? that way in a few months when we need to add changelog we dont have to refactor this as much

Reply:
Changelog is too different (field display, components, search, need to add in the textual loop to list all quarterly releases/updates), it might just need to be a completely different search method so a method rename is possible. Some of the processing methods would be reusable for changelog though. 

What abstraction did you have in mind?

Reply:
I was thinking a generic `search` method, and then a more custom "security vulnerability" and "changelog" method would call the generic jql search method.

Reply:
Added a separate commit and broke it out. Let me know if you think we need other changes. (Had to update the query too since Tiffany confirmed what we were searching in keywords.)